src/DefaultBuilder/src/StaticWebAssets/StaticWebAssetsHelper.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Xml.Linq;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore
+{
+    internal static class StaticWebAssetsHelper
```

Comment:
gross. Can we think of better ways to organise and name this functionality?

Reply:
suggestion: `StaticWebAssetsReader` and move the file-provider logic elsewhere. I also don't think there's really a justifcation for these to be extension methods.

Reply:
Done